mne/annotations.py

```diff
@@ -813,22 +813,29 @@ def _select_annotations_based_on_description(descriptions, event_id, regexp):
 
 
 @verbose
-def events_from_annotations(raw, event_id=None, regexp=None, use_rounding=True,
-                            chunk_duration=None, verbose=None):
+def events_from_annotations(raw, event_id="auto", regexp=None,
+                            use_rounding=True, chunk_duration=None,
+                            verbose=None):
     """Get events and event_id from an Annotations object.
 
     Parameters
     ----------
     raw : instance of Raw
         The raw data for which Annotations are defined.
-    event_id : dict | callable | None
+    event_id : dict | callable | None | 'auto'
         Dictionary of string keys and integer values as used in mne.Epochs
         to map annotation descriptions to integer event codes. Only the
         keys present will be mapped and the annotations with other descriptions
         will be ignored. Otherwise, a callable that provides an integer given
         a string or that returns None for an event to ignore.
         If None, all descriptions of annotations are mapped
         and assigned arbitrary unique integer values.
+        If 'auto', use an appropriate parser if the raw instance corresponds
+        to any of the supported types (currently, only Brainvision files -
+        i.e., RawBrainVision - are supported; in this case, Stimulus events
+        will be stripped to their integer part, Response events will be mapped
+        to their integer part + 1000, and all other events dropped); otherwise
+        as None.
```

Comment:
This is going to be way too long / hard to parse. I'd do:
```
    event_id : dict | callable | None | 'auto'
        Can be:

        - **dict**: string keys mapping to and integer values as used in mne.Epochs
          to map annotation descriptions to integer event codes. Only the
          keys present will be mapped and the annotations with other descriptions
          will be ignored.
        - **callable**: takes a string input, and outputs an integer to use, or None
          to ignore.
        - **None**: All descriptions of annotations are mapped and assigned
          unique integer values starting at 1 based on the ``sorted`` annotation
          descriptions.
        - **'auto'**: use an appropriate parser if the raw instance corresponds to:

          - Brainvision annotations: stimulus events stripped to their integer part;
            Response events mapped to their integer part + 1000; all other events
            dropped.
          - All others: Behaves like None.

```

Reply:
Looks good to me.

Reply:
Don't drop other events please.